server/src/main/java/com/cloud/usage/UsageServiceImpl.java

```diff
@@ -201,7 +202,8 @@ public boolean generateUsageRecords(GenerateUsageRecordsCmd cmd) {
         }
 
         boolean isAdmin = false;
```

Comment:
```suggestion
        boolean isAdmin = _accountService.isRootAdmin(caller.getId());
```

Similarly as the other vars for domain admin / user.